controllers/fenceagentsremediation_controller_test.go

```diff
@@ -149,7 +149,14 @@ var _ = Describe("FAR Controller", func() {
 			DeferCleanup(k8sClient.Delete, context.Background(), node)
 
 			Expect(k8sClient.Create(context.Background(), underTestFAR)).To(Succeed())
-			DeferCleanup(cleanupFar(), context.Background(), underTestFAR)
+			DeferCleanup(func() {
+				Expect(cleanupFar(context.Background(), underTestFAR)).To(Succeed())
+				deleteErr := k8sClient.Get(ctx, client.ObjectKeyFromObject(underTestFAR), &v1alpha1.FenceAgentsRemediation{})
```

Comment:
I do not think that this is much different that putting the `verifyEvent` directly inside the cleanup function.
Is this necessary for a reason? Could we instead explicitly call the appropriate `verifyEvent` inside the test body?

Reply:
I am confused. From your [last comment](https://github.com/medik8s/fence-agents-remediation/pull/125#discussion_r1465009887) I understood that you prefer that I won't "hide" a check (`verifyEvent(...utils.EventMessageRemoveRemediationCR)`) in a function, so I took it out of from the function.
>I do not think that this is much different that putting the verifyEvent directly inside the cleanup function.

I think that this way is clearer for showing that we test this _verifyEvent_ after cleaning the CR.

>Is this necessary for a reason? Could we instead explicitly call the appropriate verifyEvent inside the test body?

Normally we don't verify that the CR was deleted using this event, but it looks like a good sign. We can create a test that verifies that if a CR was created and when it is deleted then the finalizer was deleted as well. Would it make sense to follow this logic or leave it as a "step" in _DeferCleanup_ or even as one line in _cleanupFar_ as I introduced in the beginning?

Reply:
> I am confused.

I am sorry, I should have been more clear. I was concerned by the call to `verifyEvent` not being in the Test body as other `verifyEvent` calls are. So having it in the cleanup function or called after the cleanup function, but still not in the Test body looks the same to me.

> We can create a test that verifies that if a CR was created

that's too much logic in a test (that should be simple to read). My idea was that inside each Test body you should know if the CR is created or not, so you can explicitly call `verifyEvent` or not, without any specific check. However, I was not sure whether this is easy to do or not, that's why I asked if there was a specific reason not to do so.

Reply:
we can deal with this in a next PR, not urgent to fix it here